nupic/research/temporal_memory.py

```diff
@@ -53,30 +53,24 @@ def __init__(self,
     """
-    @param columnDimensions          (list)  Dimensions of the column space
-    @param cellsPerColumn            (int)   Number of cells per column
-    @param activationThreshold       (int)   If the number of active connected synapses on a segment is at least this threshold, the segment is said to be active.
-    @param initialPermanence         (float) Initial permanence of a new synapse.
-    @param connectedPermanence       (float) If the permanence value for a synapse is greater than this value, it is said to be connected.
-    @param minThreshold              (int)   If the number of synapses active on a segment is at least this threshold, it is selected as the best matching cell in a bursting column.
-    @param maxNewSynapseCount        (int)   The maximum number of synapses added to a segment during learning.
-    @param permanenceIncrement       (float) Amount by which permanences of synapses are incremented during learning.
-    @param permanenceDecrement       (float) Amount by which permanences of synapses are decremented during learning.
-    @param predictedSegmentDecrement (float) Amount by which active permanences of synapses of previously predicted but inactive segments are decremented.
-    @param seed                      (int)   Seed for the random number generator.
+    @param layer                     (Layer)  The cortical layer which will be processed.
+    @param activationThreshold       (int)    If the number of active connected synapses on a segment is at least this threshold, the segment is said to be active.
+    @param initialPermanence         (float)  Initial permanence of a new synapse.
+    @param connectedPermanence       (float)  If the permanence value for a synapse is greater than this value, it is said to be connected.
+    @param minThreshold              (int)    If the number of synapses active on a segment is at least this threshold, it is selected as the best matching cell in a bursting column.
+    @param maxNewSynapseCount        (int)    The maximum number of synapses added to a segment during learning.
+    @param permanenceIncrement       (float)  Amount by which permanences of synapses are incremented during learning.
+    @param permanenceDecrement       (float)  Amount by which permanences of synapses are decremented during learning.
+    @param predictedSegmentDecrement (float)  Amount by which active permanences of synapses of previously predicted but inactive segments are decremented.
+    @param seed                      (int)    Seed for the random number generator.
     """
-    # Error checking
-    if not len(columnDimensions):
-      raise ValueError("Number of column dimensions must be greater than 0")
-
-    if not cellsPerColumn > 0:
-      raise ValueError("Number of cells per column must be greater than 0")
 
     # TODO: Validate all parameters (and add validation tests)
+    if layer is None:
+      raise ValueError("Layer must be an instance of Layer class")
```

Comment:
Message is "Layer must be an instance of Layer class", but not asserting on type.  It's possible for `layer` to be something else (not a `Layer`) and pass this check.